lib/puppet/functions/deprecation.rb

```diff
@@ -8,15 +8,17 @@
   end
 
   def deprecation(key, message)
+    stacktrace = Puppet::Pops::PuppetStack.stacktrace()
+    output_message = message + ' - File and Line: ' + stacktrace[0].inspect
```

Comment:
re-use `message` here, saves you editing below.

Also, please format the message as `"${message} at ${stacktrace[0][0]}:${stacktrace[0]:[1]}"` to match common conventions on error reporting.